workers/work.go

```diff
@@ -0,0 +1,70 @@
+package workers
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/kirikami/go_db_extract/config"
+	"github.com/kirikami/go_db_extract/database"
+	"github.com/kirikami/go_db_extract/services"
+)
+
+type Work struct {
+	Function  func()
+	Completed bool
+}
+
+func NewWork(function func()) *Work {
+
+	return &Work{function, false}
+
+}
+
+func Worker(in chan *Work, out chan *Work) {
+
+	for {
+		work := <-in
+		work.Function()
+		work.Completed = true
+		out <- work
+	}
+
+}
+
+func DbWork(configs *config.TomlConfig) {
+	pending := make(chan *Work)
+	done := make(chan *Work)
+	errors := make(chan error)
+	result := make(chan services.Result, 1)
+
+	databases := len(configs.Database)
+
+	for _, config := range configs.Database {
```

Comment:
Switch this and next loops. Then you will able to remove goroutines in this loop.
Logic is simple: start workers first and then send them work.